api/main_endpoints/util/userHelpers.js

```diff
@@ -118,9 +117,13 @@ async function registerUser(userToAdd) {
     }
 
     await newUser.save()
-      .catch(_ => {
+      .catch(e => {
         result.userSaved = false;
-        result.message = 'Username already exists.';
+        if (e.code === 11000) {
+          result.message = 'Username already exists.';
+        } else {
+          result.message = 'Failed creating account.';
+        }
```

Comment:
can we also use `logger.error` to log whatever the error is
```suggestion
        result.message = 'Failed creating account.';
        if (e.code === 11000) {
          result.message = 'Username already exists.';
        }
```